docs_source_files/content/support_packages/mouse_and_keyboard_actions_in_detail.de.md

```diff
@@ -24,7 +24,7 @@ IWebElement e = driver.FindElement(By.Id("testElement"));
 # We don't have a Ruby code sample yet -  Help us out and raise a PR  
   {{< / code-panel >}}
   {{< code-panel language="javascript" >}}
-// We don't have a JavaScript code sample yet -  Help us out and raise a PR  
+let e = await driver.findElement(By.id("testElement"));
```

Comment:
As far as I remember Await may not be required to find an element. can you please check ?
it can just be let e = driver.findElement(By.id("testElement"));

Reply:
Hi @manoj9788,

Thanks for the review.

As findElement returns a  promise(!webdriver.WebElementPromise), i have used await keyword. 

Please let me know if am wrong.

Reply:
If you look at WebElement and Alert in lib/webdriver.js, you'll see there are also WebElementPromise and AlertPromise classes. These basically wrap the "promised" value in the user-facing API. So all of the commands will internally resolve the promise and execute the command as if the user did it manually.

I don't have access to my test machine - but if you have an example code you can try both ways(with and without **Await**) and observe, it should work without Await is my understanding.

Reply:
Yes @manoj9788 ,

It works without await too. I have looked into the core implementation of WebElementPromise. as you said, yes WebElement is resolving promise internally. Thanks for the explanation. I will change the code accordingly.

but, i was wondering about the discussion in the below link

https://github.com/SeleniumHQ/selenium/issues/2969

It says promise manager is removed and user has to resolve promises externally, i was confused about precise understanding of placing async/await, because there are few promises which is internally resolved and few command api requires async/wait
(or) may be can i assume, the async await is only required for core selenium api like (get, navigate, click ,sendKeys)

Please help me in understanding this, i would be thankful and it will help me in future PRs.

Thanks!

Reply:
Thanks for confirming :)

Even without the promise manager, there are a few parts in the API that smooth over the async bits so you don't have to handle every promise. Mainly this is when creating/fetching a WebDriver, WebElement, or Alert instance.

I gave a talk at Selenium Conference Austin 2017: Here are the slides https://www.slideshare.net/ManojKumar103/async-webdriverjs

Reply:
Great, i will look into the slides.

Lot of learning outcomes in one PR. This made my day. Thanks @manoj9788 

I've updated the code accordingly.

Reply:
Glad it's helpful! 
You are doing great work! Keep continuing and inspire others!